spec/apis/single-spa-events-api.spec.js

```diff
@@ -0,0 +1,165 @@
+export default function singleSpaEventsApi() {
+	const dummyApp = {
+		bootstrap() {return Promise.resolve()},
+		mount() {return Promise.resolve()},
+		unmount() {return Promise.resolve()},
+		unload() {return Promise.resolve()},
+	};
+
+	describe(`events api :`, () => {
+		beforeAll(() => {
+			singleSpa.declareChildApplication('russell', () => Promise.resolve(dummyApp), () => window.location.hash.indexOf("#/russell") === 0);
+		});
+
+		describe(`single-spa:routing-event`, () => {
+			beforeEach(() => {
+				window.location.hash = ``;
+			});
+
+			it(`is fired on the window whenever the hash changes`, done => {
+				singleSpa
+				.triggerAppChange() // start with a clean slate (no previous tests doing anything)
+				.then(() => {
+					window.addEventListener("single-spa:routing-event", finishTest);
+					window.location.hash = `#/hash-was-changed`;
+
+					function finishTest() {
+						window.removeEventListener("single-spa:routing-event", finishTest);
+						done();
+					}
+				})
+				.catch(err => {
+					fail(err);
+					done();
+				});
+			});
+
+			it(`is fired on the window whenever the url changes`, done => {
+				singleSpa
+				.triggerAppChange() // start with a clean slate (no previous tests doing anything)
+				.then(() => {
+					window.addEventListener("single-spa:routing-event", finishTest);
+					window.location.hash = `#/push-state-called`;
```

Comment:
Good point -- I mean to call pushState instead of changing the hash.

Will fix